agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -122,39 +120,38 @@ func UpdateTaskProtectionHandler(state dockerstate.TaskEngineState,
 			Tasks:             aws.StringSlice([]string{task.Arn}),
 		})
 
-		logger.Debug("updateTaskProtection response:", logger.Fields{
-			loggerfield.TaskProtection: response.ProtectedTasks,
-			loggerfield.Error:          err,
-			loggerfield.Reason:         response.Failures,
-		})
-
 		if err != nil {
 			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
 			logger.Info(fmt.Sprintf("ECS throws an exception with type %s", exceptionType))
 			writeJSONResponse(w, statusCode, err.Error(), updateTaskProtectionRequestType)
 			return
 		}
 
+		logger.Debug("updateTaskProtection response:", logger.Fields{
+			loggerfield.TaskProtection: response.ProtectedTasks,
+			loggerfield.Reason:         response.Failures,
+			loggerfield.Error:          err,
```

Comment:
`err` will always be `nil` so maybe we should not include it here to avoid any confusions.